File: service/src/main/java/coding/challenge/configuration/TempPathProperties.java
package coding.challenge.configuration;

import lombok.Getter;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

/**
 * Represents configuration properties for temporary file paths.
 */
@ConfigurationProperties(prefix = "file.temp.path")
@Getter
public class TempPathProperties {

    /**
     * Directory for audio files
     */
    private final Path audioFiles;

    public TempPathProperties(String audioFiles) throws IOException {
        this.audioFiles = Path.of(audioFiles);
        Files.createDirectories(this.audioFiles);
    }

}

File: service-kt/src/main/kotlin/coding/challenge/configuration/TempPathProperties.kt
package coding.challenge.configuration

import org.springframework.boot.context.properties.ConfigurationProperties
import java.nio.file.Files
import java.nio.file.Path

/**
 * Represents configuration properties for temporary file paths.
 */
@ConfigurationProperties(prefix = "file.temp.path")
class TempPathProperties(audioFiles: String) {

    /**
     * Directory for audio files
     */
    val audioFiles: Path = Path.of(audioFiles)

    init {
        Files.createDirectories(this.audioFiles)
    }

}
